app/main.py

```
@@ -1,2 +1,79 @@
+        while True:
+            if self.table[index] == []:
+                self.table[index] = [key, value, hashed_key]
+                self.length += 1
+                break
+            if self.table[index][2] == hashed_key\
+                    and self.table[index][0] == key:
+                self.table[index][1] = value
+                break
+            index = (index + 1) % self.capacity
+        if self.length == self.load:
+            self.resize()
+
+    def resize(self):
+        self.capacity *= 2
+        self.length = 0
+        self.load = int(self.capacity * 2 / 3)
+        prep_table = self.table
+        self.table = [[] for _ in range(self.capacity)]
+        for element in prep_table:
+            if len(element):
+                self.__setitem__(element[0], element[1])
+
+    def __len__(self):
+        return self.length
+
+    def __getitem__(self, key):
+        index = hash(key) % self.capacity
+        element = self.table[index]
+        while element:
+            if element[2] == hash(key) and element[0] == key:
+                return element[1]
+            index = (index + 1) % self.capacity
+            element = self.table[index]
+        raise KeyError()
+
+    def clear(self):
+        self.capacity = 8
+        self.length = 0
+        self.load = int(self.capacity * 2 / 3)
+        self.table = [[] for _ in range(self.capacity)]
+
+    def __delitem__(self, key):
+        index = hash(key) % self.capacity
+        element = self.table[index]
+        while element:
+            if element[2] == hash(key) and element[0] == key:
+                self.table[index] = []
+                self.length -= 1
+                break
+            index = (index + 1) % self.capacity
+            element = self.table[index]
+        raise KeyError
+
+    def get(self, key, value=None):
+        list_actual_keys = (_[0] for _ in self.table if _ != [])
+        if key not in list_actual_keys:
+            return value
```

You should raise an error if key doesn't present and user didn't pass the default value

---

This method returns the value for the given key. If the key is not available, the default value None is returned. Not?